File: RxKit/src/main/java/com/tamsiree/rxtool/RxPackageManagerTool.java
package com.tamsiree.rxtool;

import android.content.Context;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;

import java.util.ArrayList;
import java.util.List;

/**
 * @author tamsiree
 * @date 2018/5/2 15:00:00
 */
public class RxPackageManagerTool {

    public static List<PackageInfo> initPackageManager(Context mContext) {
        PackageManager mPackageManager = mContext.getPackageManager();
        return mPackageManager.getInstalledPackages(0);
    }

    /**
     * 包名是否存在
     * @param mContext 实体
     * @param packageName 待检测 包名
     * @return 结果
     */
    public static boolean haveExistPackageName(Context mContext,String packageName) {
        List<PackageInfo> packageInfos =  initPackageManager(mContext);
        List<String> mPackageNames = new ArrayList<>();
        if (packageInfos != null) {
            for (int i = 0; i < packageInfos.size(); i++) {
                mPackageNames.add(packageInfos.get(i).packageName);
            }
        }
        return mPackageNames.contains(packageName);
    }
}

File: RxKit/src/main/java/com/tamsiree/rxkit/RxPackageManagerTool.kt
package com.tamsiree.rxkit

import android.content.Context
import android.content.pm.PackageInfo
import java.util.*

/**
 * @author tamsiree
 * @date 2018/5/2 15:00:00
 */
object RxPackageManagerTool {
    @JvmStatic
    fun initPackageManager(mContext: Context): List<PackageInfo> {
        val mPackageManager = mContext.packageManager
        return mPackageManager.getInstalledPackages(0)
    }

    /**
     * 包名是否存在
     * @param mContext 实体
     * @param packageName 待检测 包名
     * @return 结果
     */
    @JvmStatic
    fun haveExistPackageName(mContext: Context, packageName: String): Boolean {
        val packageInfos = initPackageManager(mContext)
        val mPackageNames: MutableList<String> = ArrayList()
        for (i in packageInfos.indices) {
            mPackageNames.add(packageInfos[i].packageName)
        }
        return mPackageNames.contains(packageName)
    }
}